### DISCUSSION TITLE
OCI runtime create failed due to incompatible CNI versions

### DISCUSSION THREAD

**Discussion by AbleHou** [containerd/nerdctl#1129] [Q&A] [answered]
### Description

nerdctl run fail

### Steps to reproduce the issue

1.
2.
3.

### Describe the results you received and expected

I hope container STATUS is Up but:
FATA[0003] OCI runtime create failed: container_linux.go:370: starting container process caused: process_linux.go:459: container init caused: Running hook #0:: error running hook: exit status 1, stdout: , stderr: time="2022-05-11T10:09:33+08:00" level=fatal msg="failed to call cni.Setup: plugin type=\"bridge\" failed (add): incompatible CNI versions; config is \"1.0.0\", plugin supports [\"0.1.0\" \"0.2.0\" \"0.3.0\" \"0.3.1\" \"0.4.0\"]"

### What version of nerdctl are you using?

nerdctl-0.19.0-linux-amd64

**Comment by junnplus** [comment]
What’s your CNI bridge plugin version?

**Comment by fahedouch** [comment]
Plz download the latest plugin version https://github.com/containernetworking/plugins/releases/tag/v1.1.1 and do not build plugin  bin in local I faced the same issue when building plugin in local

**Comment by AbleHou** [comment]
@junnplus CNI bridge plugin v1.1.1, but I solved it by "nerdctl run --network host xxxxx"

**Comment by AbleHou** [comment, marked as answer]
solution:
plugin: https://github.com/containernetworking/plugins/releases/tag/v1.1.1  
nerdctl: nerdctl version 0.20.0

**Comment by omegazeng** [comment]
```
[root@localhost mysql]# nerdctl version
Client:
 Version:       v0.20.0
 OS/Arch:       linux/amd64
 Git commit:    e77e05b5fd252274e3727e0439e9a2d45622ccb9

Server:
 containerd:
  Version:      v1.6.6
  GitCommit:    10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
[root@localhost mysql]# /opt/cni/bin/bridge -v
CNI bridge plugin v1.1.1
[root@localhost mysql]# nerdctl compose up -d
INFO[0000] Creating network mysql_mysql
INFO[0000] Creating network mysql_default
WARN[0000] Ignoring: volume: Bind: [CreateHostPath]     
INFO[0000] Ensuring image registry.cn-beijing.aliyuncs.com/laiye-poc/mariadb:10.5 
INFO[0000] Creating container mysql_mysql_1
FATA[0000] failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: time="2022-06-15T19:42:45+08:00" level=fatal msg="failed to call cni.Setup: plugin type=\"bridge\" failed (add): incompatible CNI versions; config is \"1.0.0\", plugin supports [\"0.1.0\" 
\"0.2.0\" \"0.3.0\" \"0.3.1\" \"0.4.0\"]"
Failed to write to log, write /var/lib/nerdctl/1935db59/containers/default/5df399668e18e470b43adc64c9eddea445c131f7f30d960769821804de125900/oci-hook.createRuntime.log: file already closed: unknown
FATA[0000] error while creating container mysql_mysql_1: exit status 1 
```
issue is not resolved yet
@AbleHou What version of containerd are you using?

**Comment by junnplus** [comment]
@omegazeng Default cni path is `/usr/lib/cni`, using `nerdctl --cni-path=/opt/cni/bin` to change it .

**Reply by fahedouch** [reply]
@omegazeng or the env var `CNI_PATH`

**Comment by omegazeng** [comment]
> @omegazeng Default cni path is `/usr/lib/cni`, using `nerdctl --cni-path=/opt/cni/bin` to change it .
It works! Thx.
